web/components/templates/playground/components/PlaygroundMessagesPanel.tsx

```diff
@@ -50,6 +50,31 @@ const PlaygroundMessagesPanel = ({
   const resizeTimeoutRef = useRef<NodeJS.Timeout>();
   const clientHeightRef = useRef<number>(0);
 
+  useEffect(() => {
+    if (mappedContent?.schema.request?.messages) {
+      const messages = mappedContent.schema.request.messages;
+      const needsIds = messages.some((message) => !message.id);
+
+      if (needsIds) {
+        const messagesWithIds = messages.map((message, index) => ({
+          ...message,
+          id: message.id || `msg-${index}-${Date.now()}`,
+        }));
+
+        setMappedContent({
+          ...mappedContent,
+          schema: {
+            ...mappedContent.schema,
+            request: {
+              ...mappedContent.schema.request,
+              messages: messagesWithIds,
+            },
+          },
+        });
+      }
+    }
+  }, [mappedContent, setMappedContent]);
+
```

Comment:
this code seems a bit wrong - will lead to an infinite loop (the effect is running on a mappedContent dependency and we're setting mapped content inside it)

should use `setMappedContent(prev => ...)` instead